src/redux/sagas/extensions/extensionDeprecate.ts

```diff
@@ -27,13 +27,15 @@ function* extensionDeprecate({
 
     yield takeFrom(txChannel, ActionTypes.TRANSACTION_CREATED);
 
-    yield put<AllActions>({
-      type: ActionTypes.EXTENSION_DEPRECATE_SUCCESS,
-      payload: {},
-      meta,
-    });
-
-    yield waitForTxResult(txChannel);
+    const result = yield waitForTxResult(txChannel);
+
+    if (result.type === ActionTypes.TRANSACTION_SUCCEEDED) {
```

Comment:
Here we wait for `waitForTxResult`, which is listening for error, success, and cancelled. 

If success, then we emit saga's success message. `waitForTxResult` throws if there's a tx error, which triggers the saga's catch block.

Reply:
This made me think, there are also some sagas (including expenditure sagas) that are using such approach instead of `waitForTxResult`:
```
yield takeFrom(createDomain.channel, ActionTypes.TRANSACTION_SUCCEEDED);

// put the success action
```
It looks like they need to be switched over to use `waitForTxResult` in order to properly handle errors too.

Reply:
https://github.com/JoinColony/colonyCDapp/issues/1018

Reply:
Yep, I'll make a note that this contributes to that issue